### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by rubydevi**
## Description
In this project, I started building a school library app. In this initial step, I implemented the classes to represent `students` and `teachers`.

### Changes made
- Created class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameters. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if a person is of age or if they have permission from parents.
- Created class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Created class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class are in a separate file.

_Thanks for your feedback in advance_ 😄

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
```
Comment (DammyShittu) [inline student.rb]: - [x] Kindly endeavor to be consistent in your definitions. The `name` parameter in Person class uses colon (:), but here you are using equal sign (=).

- [x] What you defined in the Person class is `parents_permission`, "parents" with an "s".

- [x] The `name` in the super arguments should be assigned correctly.

  ```
  def initialize(classroom, age, name: 'Unknown', parents_permission: true)
    super(age, name: name, parents_permission: parents_permission)
  ```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    @name = name
+    @age = age
+    @parents_permission = parents_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def can_use_services?
+    @parent_permission || of_age?
```
Comment (DammyShittu) [inline person.rb]: - [ ] This should be `@parents_permission`.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @rubydevi,

## Highlights
✔️ Private method

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (rubydevi) [reply student.rb]: Thanks, @DammyShittu for pointing that out I completely forgot to recheck (`ctrl` + `z`) after making some changes to an already working code. Lesson learned 💯

**Review by prolajumokeoni — APPROVED** [review]
Hi @rubydevi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - Changes requested have been implemented✅ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**